### DISCUSSION TITLE
Adaptive mesh refinement

### DISCUSSION THREAD

**Discussion by tqcthai** [idaholab/moose#25680] [Q&A General] [answered]
Hello Moose community,

I am studying adaptive mesh with Moose. There are several indicators we can use like Jump Indicator, Gradient Jump Indicator of a variable. However, I want to study with the simplest one, the indicator is the variable itself, the element will be refined when its value higher than a threadhold. I wonder how to do it? I've tried with Moose variable like the code below but it doesn't work.
```
[Adaptivity]
  marker = errorfrac
  steps = 1
  max_h_level = 4
  [Indicators]
    [error]
      type = MooseVariable       #GradientJumpIndicator or ValueJumpIndicator
      variable = c 
      scale_by_flux_faces = true
    []
  []
  [Markers]
    [errorfrac]
      type = ErrorFractionMarker
      indicator = error
      refine = 0.5 #cut value
      coarsen = 0
    []
  []
[]
```

**Comment by GiudGiud** [comment]
Did you try specifying no indicator block and instead passing the name of the variable to the parameter. (The variable being defined normally as an auxvariable or Variable as usual)

**Reply by GiudGiud** [reply]
I have not tried, but something like this maybe?
```
[Adaptivity]
  marker = errorfrac
  steps = 1
  max_h_level = 4
  [Markers]
    [errorfrac]
      type = ErrorFractionMarker
      indicator = error
      refine = 0.5 #cut value
      coarsen = 0
    []
  []
[]

[AuxVariables]
  [error]
  []
[]
```

**Reply by tqcthai** [reply, marked as answer]
It doesn't work. Anyway, I found out a way to define it as an analytical solution of a function 
```

[Functions]
  [solution]
    type = ParsedFunction
    expression = x
  []
[]

[Adaptivity]
  marker = errorfrac
  steps = 1
  max_h_level = 4
  [Indicators]
    [error]
      type = AnalyticalIndicator
      variable = c
      function = solution
    []
  []
  [Markers]
    [errorfrac]
      type = ErrorFractionMarker
      coarsen = 0.0
      indicator = error
      refine = 0.1
    []
  []
[]
```
